pyroengine/engine/predictor.py

```diff
@@ -3,29 +3,24 @@
 # This program is licensed under the Apache License version 2.
 # See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
 
-import torch
+from pyroengine.models.yolov5 import Yolo_v5
 
 
 class PyronearPredictor:
     """This class use the last pyronear model and run our smoke detection model on it
     Examples:
-        >>> pyronearPredictor = PyronearPredictor()
+        >>> pyronearPredictor = PyronearPredictor(model_weights=/path/to/model.onnx)
         >>> im = Image.open("image.jpg")
         >>> res = pyronearPredictor.predict(im)
     """
 
     def __init__(self, model_weights: str = None, conf: float = 0.25):
         """Init predictor"""
         # Model definition
-        if model_weights is None:
-            model_weights = "https://github.com//pyronear//pyro-vision//releases//download//v0.1.2//yolov5s_v001.onnx"
-        self.model = torch.hub.load(
-            "ultralytics/yolov5", "custom", path=model_weights, force_reload=True
-        )  # local model
-        self.model.conf = conf
+        self.model = Yolo_v5(model_weights=model_weights, conf_thres=conf)
 
     def predict(self, im):
         """Run prediction"""
-        pred = self.model(im)
+        pred = self.model.forward(im)
 
-        return pred.xyxy[0].numpy()
+        return pred
```

Comment:
According to the constructor and the call method implementation, this class does nothing that `Yolo_v5` doesn't

I'd suggest directly instantiating a `Yolo_v5`